src/NuGet.Core/NuGet.Protocol/Model/IgnoreCaseStringKeyDictionary.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using NuGet.Protocol.Converters;
+
+namespace NuGet.Protocol.Model
+{
+    [JsonConverter(typeof(VulnerabilityFileDataConverter))]
+    internal class VulnerabilityFileData : IReadOnlyDictionary<string, IReadOnlyList<VulnerabilityInfo>>
+    {
+        private IReadOnlyDictionary<string, IReadOnlyList<VulnerabilityInfo>> _inner;
+
+        public VulnerabilityFileData(Dictionary<string, IReadOnlyList<VulnerabilityInfo>> inner)
+        {
+            _inner = inner ?? throw new ArgumentNullException(paramName: nameof(inner));
+
+            if (inner.Comparer != StringComparer.InvariantCultureIgnoreCase)
+            {
+                throw new ArgumentException(
+                    message: "Dictionary does not use " + nameof(StringComparer.InvariantCultureIgnoreCase) + " as the comparer",
+                    paramName: nameof(inner));
+            }
+        }
+
+        public IReadOnlyList<VulnerabilityInfo> this[string key] => _inner[key];
+
+        public IEnumerable<string> Keys => _inner.Keys;
+
+        public IEnumerable<IReadOnlyList<VulnerabilityInfo>> Values => _inner.Values;
+
+        public int Count => _inner.Count;
+
+        public bool ContainsKey(string key) => _inner.ContainsKey(key);
+
+        public IEnumerator<KeyValuePair<string, IReadOnlyList<VulnerabilityInfo>>> GetEnumerator() => _inner.GetEnumerator();
+
+        public bool TryGetValue(string key, out IReadOnlyList<VulnerabilityInfo> value) => _inner.TryGetValue(key, out value);
+
+        IEnumerator IEnumerable.GetEnumerator() => _inner.GetEnumerator();
+    }
```

Comment:
```suggestion
    internal class VulnerabilityFileData : Dictionary<string, IReadOnlyList<VulnerabilityInfo>>, IReadOnlyDictionary<string, IReadOnlyList<VulnerabilityInfo>>
    {
        public VulnerabilityFileData()
          : base(StringComparer.OrdinalIgnoreCase)
        {
        }
    }
```

Not 100% sure this will work but you can try having this class inherit from `Dictionary<string, IReadOnlyList<VulnerabilityInfo>>` then call the base constructor with `StringComparer.OrdinalIgnoreCase` making this class always be a case-insensitive dictionary and not needing to have an inner dictionary.  That could also mean that you would not need a custom JSON converter?

Reply:
This worked well, thanks!

I also made a slight change, making it a generic type `CaseInsenstiveDictionary<TValue>`, so theoretically it could be re-used if it ends up being useful, but we no longer need our own JsonConverter.